packages/dds/merge-tree/src/segmentGroupCollection.ts

```diff
@@ -21,9 +21,16 @@ export class SegmentGroupCollection {
         return this.segmentGroups.empty();
     }
 
-    public enqueue(segmentGroup: SegmentGroup) {
+    public enqueue(segmentGroup: SegmentGroup, sourceSegment?: ISegment) {
```

Comment:
since this is only used internally in copy, maybe make a private enqueue that takes the segment, and call that from copy, and from the public enqueue so we don't need a signature change